src/App.tsx

```diff
@@ -1,24 +1,149 @@
+      userId: USER_ID, title: query.trim(), completed: false, id: 0,
+    });
+
+    TodoServices.postTodos(
+      { userId: USER_ID, title: query.trim(), completed: false },
+    )
+      .then(newItem => {
+        setTodos(currentTodos => [...currentTodos, newItem]);
+        setQuery('');
+      })
+      .catch(() => {
+        setError(ErrorType.AddTodoError);
+      })
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const deletePost = (todoId: number) => {
+    TodoServices.deleteTodos(todoId);
+    setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+  };
+
+  const filteredTodos = useMemo(() => {
+    switch (filterValue) {
+      case FilterValue.Active:
+        return todos.filter(todo => !todo.completed);
+      case FilterValue.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, filterValue]);
+
+  const handleFilterChange = (filter: FilterValue) => {
+    setFilterValue(filter);
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const clearCompleted = () => {
+    const todosCompleted = todos.filter(todo => todo.completed);
+
+    const todosIdCompleted = todosCompleted.map(todo => todo.id);
+
+    todosIdCompleted.forEach(item => {
+      TodoServices.deleteTodos(item)
+        .catch(() => {
+          setError(ErrorType.DeleteTodoError);
+        });
+    });
+
+    const newTodos = todos.filter(todo => !todo.completed);
+
+    setTodos(newTodos);
+  };
+
+  const handleToggleAll = async () => {
+    try {
+      const updatePromises = todos.map(async (todo) => {
+        const updatedTodo = { ...todo, completed: !todo.completed };
+
+        await TodoServices.updateTodos(updatedTodo);
+
+        return updatedTodo;
+      });
+
+      const updatedTodos = await Promise.all(updatePromises);
+
+      setTodos(updatedTodos);
+    } catch (errorR) {
+      setError(ErrorType.UpdateTodoError);
+    }
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          onSubmit={addTodo}
+          USER_ID={USER_ID}
+          setError={setError}
+          tempTodo={tempTodo}
+          query={query}
+          setQuery={setQuery}
+          handleToogleAll={handleToggleAll}
+        />
+
+        <TodoList
+          todos={filteredTodos}
+          deletePost={deletePost}
+          setTodos={setTodos}
+          setError={setError}
+        />
+
+        {/* Hide the footer if there are no todos */}
```

Comment:
Remove comment. Try it write obvious code that explains all we need without comments)